### PR TITLE
Fix: Logging configs not being set

### PR DESCRIPTION
**Pull request by ibraheem-abe**
Please see this for more context:
https://github.com/opentensor/KnowledgeBase/issues/30#issuecomment-2187469983

### CODE REVIEW

**Review by thealligatorking — APPROVED** [review]

File: bittensor/btlogging/loggingmachine.py [hunk 1]
```diff
@@ -70,19 +70,19 @@ class LoggingMachine(StateMachine):
...
     )
 
-    disable_trace: Default = Trace.to(Default)
+    disable_trace = Trace.to(Default)
 
-    disable_debug: Default = Debug.to(Default)
+    disable_debug = Debug.to(Default)
 
-    disable_logging: Disabled = (
+    disable_logging = (
```
Comment (thewhaleking) [inline bittensor/btlogging/loggingmachine.py:85]: Why are we removing type hints?

File: bittensor/btlogging/loggingmachine.py [hunk 2]
```diff
@@ -107,14 +107,31 @@ def __init__(self, config: bittensor.config, name: str = BITTENSOR_LOGGER_NAME):
...
         # set up all the loggers
         self._logger = self._initialize_bt_logger(name)
         self.disable_third_party_loggers()
+        self._enable_initial_state(self._config)
+
+    def _enable_initial_state(self, config):
+        """Set correct state action on initializing"""
+        if config.trace:
+            self.enable_trace()
+        elif config.debug:
```
Comment (thewhaleking) [inline bittensor/btlogging/loggingmachine.py:124]: This `elif` would only enable debug if trace is not enabled. Is that what we want?

File: bittensor/btlogging/loggingmachine.py (hunk 1 above)
Comment (ibraheem-abe) [reply bittensor/btlogging/loggingmachine.py:85]: I think the type hints here are not required as, for example, 'Debug' is not a type here; instead it is an instance of the State class in the Statemachine. 
The IDE also warns with the error
`Variable not allowed in type expression`

File: bittensor/btlogging/loggingmachine.py (hunk 2 above)
Comment (ibraheem-abe) [reply bittensor/btlogging/loggingmachine.py:124]: Yes, that is the intended behavior. At a time, we can enable 1 exclusive logging state—Trace, Debug, or Default—based on the configuration provided.

**Trace**: This is the most verbose level, logging all messages including trace level messages. This has a level of 5. It contains all logging levels i.e. trace + debug + default

**Debug**: This has a level of 10. It contains debug + default logging levels. 

**Default**: This has a level of 20. It only contains default levels i.e. info, error, warning, success

**Review by thewhaleking — APPROVED** [review]

**Pull request merged**